### PR TITLE
BRASS 639: updating documentation

### PR DESCRIPTION
**Pull request by PingDavidR**
- added a TODO document
- added file system explanation
- added notes on specifics of the provider
- other edits

### CODE REVIEW

File: contributing/development.md [hunk 1]
```diff
@@ -35,8 +40,8 @@ The PingDirectory Terraform provider relies on the [PingDirectory Go Client](htt
 
 If changes are needed in the Go client, the `replace` command in the `go.mod` file can be used to point to a modified local Go client while testing.
 
-```
-replace github.com/pingidentity/pingdirectory-go-client v0.0.0 => ../pingdirectory-go-client
+```txt
+replace github.com/pingidentity/pingdirectory-go-client v0.0.0 with ==> ./pingdirectory-go-client
```
Comment (henryrecker-pingidentity) [inline contributing/development.md:44]: Remove the "with" here - this line is a code line that is expected in the go.mod file. https://go.dev/ref/mod#go-mod-file-replace

File: contributing/filelayout.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		serverinstance.NewProxyServerInstanceResource,
+		serverinstance.NewSyncServerInstanceResource,
+		trustmanagerprovider.NewBlindTrustManagerProviderResource,
+		trustmanagerprovider.NewJVMDefaultTrustManagerProviderResource,
+		trustmanagerprovider.NewFileBasedTrustManagerProviderResource,
+		trustmanagerprovider.NewThirdPartyTrustManagerProviderResource,
+	}
+```
+
+The **utils** package contains functions used across other packages.
```
Comment (henryrecker-pingidentity) [inline contributing/filelayout.md:29]: This is out of date - there isn't any utils package any more. Things have been split into operations and types packages.

File: contributing/filelayout.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+Finally, the actual **configuration** object resources are contained under the **resource/** folder:
+- **resource/ldap/** contains the non-production-ready resource for managing LDAP users
```
Comment (henryrecker-pingidentity) [inline contributing/filelayout.md:32]: This no longer exists

File: contributing/filelayout.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+- **resource/config/** contains the configuration object resources
+
+The **config** folder will have two types of resources:
+- Resources that only support a single type, such as **Location** and **Global Configuration**.  These resources will be found directly in the **config** folder.
+- Resources with an API that manages multiple types, such at the **Trust Manager**.  PingDirectory supports multiple types of providers (Blind Trust, Third Party, JVM Default, and File Based).  In these cases, each type is located in a separate sub-package folder that is named for the type.
+
+## Tests
```
Comment (henryrecker-pingidentity) [inline contributing/filelayout.md:39]: I would label this "Acceptance Tests"

File: contributing/filelayout.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+│   │           ├── jvm_default_trust_manager_provider_resource.go
+│   │           └── third_party_trust_manager_provider_resource.go
+│   └── types                    ← utility type stuff
+│       ├── conversion.go        ← this file might need updating (see below)
+│       ├── definitions.go
+│       └── utils.go
+└── main.go                      ← standard Go convention
+```
+
+#### Note on conversions.go:
```
Comment (henryrecker-pingidentity) [inline contributing/filelayout.md:88]: With the MR I put in this morning, there's no longer a need to update this file. All the generated methods for all the enums are in the client, not the provider.

File: TODO.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# TODO
+
+Items for improvement
+
+### To Do
+
+- [ ] Investigate splitting the utils package out into more granular packages - Ping internal ticket BRASS-630
```
Comment (henryrecker-pingidentity) [inline TODO.md:7]: This is already done

File: README_ProviderNotes.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Edit-only Resources
+
+Certain configuration objects are edit-only. These objects do not fit in the typical Terraform model. Terraform expects to manage the entire lifecycle of an object, from creation to deletion. In the case of PingDirectory, some objects are inherent, often with default values.  However, leaving these objects out of the control of Terraform would severely restrict the ability to manage PingDirectory configuration using Terraform HCL. As an example, consider the global configuration, which is very useful and often modified in PingDirectory; this resource is edit-only.
+
+Other providers have functionality to manage resources in this category. One particular example is in this AWS provider resource: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_security_group.  For items such as this and others in the registry, the strategy in Terraform is to “adopt” them on a typical create, and "forget" them on a delete action.  When Terraform forgets, the resource still exists and can be managed elsewhere. This pattern is what the PingDirectory provider has implemented for edit-only objects.
```
Comment (henryrecker-pingidentity) [inline README_ProviderNotes.md:11]: I don't think we need to discuss the AWS provider and how it handles things here.

File: README_ProviderNotes.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+While the global configuration is used as an example here, all edit-only resources are handled in this manner.
+
+### Attributes in edit-only resources
+
+As part of the edit-only strategy, all attributes of these resources are defined as **_computed_** so that users do not need to specify every available attribute.
+
+## id instead of name
+
+When running **_dsconfig_** commands, the instance is referred to using the **name** variable.  The convention in a Terraform provider is to use **id** as the naming attribute in most cases.
```
Comment (henryrecker-pingidentity) [inline README_ProviderNotes.md:23]: It'd be good to show an example of location via dsconfig versus location via terraform here to explain what this means

**Review by henryrecker-pingidentity — CHANGES_REQUESTED** [review]

File: contributing/development.md (hunk 1 above)
Comment (PingDavidR) [reply contributing/development.md:44]: heh, not sure how that go there

File: README_ProviderNotes.md (hunk 8 above)
Comment (PingDavidR) [reply README_ProviderNotes.md:23]: done

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**